### PR TITLE
user formatting for axis labels

### PR DESCRIPTION
**Pull request by cooldome**
Adds argument `labels` to `scale_x_discrete`, `scale_x_continuous`,  `scale_y_discrete`, `scale_x_continuous` that is a closure proc for custom formatting of text labels. Argument `labels` is taken from original ggplot2 for R that also accepts formatting function. I use them for outputting dates in mon-year format and adding thousands separator decimal for floating point numbers.

This PR requires changes to ginger https://github.com/Vindaar/ginger/pull/18 in order to pass tests.

### CODE REVIEW

**Comment by cooldome** [comment]
Sample plot
[test_plot_labels.pdf](https://github.com/Vindaar/ggplotnim/files/4617431/test_plot_labels.pdf)

**Comment by Vindaar** [comment]
Awesome, I'll take a look hopefully tomorrow!

**Comment by Vindaar** [comment]
Nice, thanks a lot. This is a very elegant solution!

I checked out the code locally and noticed that the recipes still need to be fixed. Could you fix the `scale_x_continuous` calls that make use of `dcKind` in there?

And feel free to add a recipe as an example for custom labels. But I can also do that after I merged this.

**Comment by cooldome** [comment]
All done. Fixed all recipes that used scale_Z_continuous(dcKind = dcDiscrete) to use scale_Z_discrete(). Added to new recipes to demo user formatting.

File: recipes/rPeriodicTable.nim [hunk 1]
```diff
@@ -72,7 +72,7 @@ ggplot(elements, aes("x", "y", fill = "metal")) +
             aes = aes(x = f{3 - tileWidth/4.0 + 0.25},
                       width = f{tileWidth / 2.0},
                       height = f{tileHeight})) +
-  scale_y_continuous(dcKind = dcContinuous) +
```
[the remark this answers is not in the document]
Comment (cooldome) [reply recipes/rPeriodicTable.nim:75]: oops, fixed

File: recipes/rPeriodicTable.nim [hunk 2]
```diff
@@ -102,7 +102,7 @@ ggplot(elements, aes("x", "y", fill = "metal")) +
   legendPosition(0.82, 0.1) +
   theme_void() +
   scale_y_reverse() +
-  scale_x_continuous(dcKind = dcContinuous) +
+  scale_x_discrete() +
```
[the remark this answers is not in the document]
Comment (cooldome) [reply recipes/rPeriodicTable.nim]: fixed

**Comment by cooldome** [comment]
I am afraid I don't have ntangle installed and I work in the environment where installing tools from internet is problematic, you need to get it from approved sources.

**Comment by Vindaar** [comment]
> I am afraid I don't have ntangle installed and I work in the environment where installing tools from internet is problematic, you need to get it from approved sources.

Hey, no worries at all! :)

**Pull request merged**